File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/03-additional/04.grades.js
function grades(input) {

    const studentsCount = Number(input[0]);
    let index = 1;

    let counterTopGrades = 0;
    let counterGradesBetween4And5 = 0;
    let counterGradesBetween3And4 = 0;
    let counterLowGrades = 0;
    let sumOfGrades = 0;

    for (let i = 1; i <= studentsCount; i++) {
        let currentGrade = Number(input[index++]);
        sumOfGrades += currentGrade;
        if (currentGrade > 4.99) {
            counterTopGrades++;
        } else if (currentGrade > 3.99) {
            counterGradesBetween4And5++;
        } else if (currentGrade > 2.99) {
            counterGradesBetween3And4++;
        } else {
            counterLowGrades++;
        }
    }

    const topGaradesPercentage = counterTopGrades / studentsCount * 100;
    const gradesBetween4And5Percentage = counterGradesBetween4And5 / studentsCount * 100;
    const gradesBetween3And4Percentage = counterGradesBetween3And4 / studentsCount * 100;
    const lowGaradesPercentage = counterLowGrades / studentsCount * 100;
    const averageGrades = sumOfGrades / studentsCount;

    console.log(`Top students: ${topGaradesPercentage.toFixed(2)}%`);
    console.log(`Between 4.00 and 4.99: ${gradesBetween4And5Percentage.toFixed(2)}%`);
    console.log(`Between 3.00 and 3.99: ${gradesBetween3And4Percentage.toFixed(2)}%`);
    console.log(`Fail: ${lowGaradesPercentage.toFixed(2)}%`);
    console.log(`Average: ${averageGrades.toFixed(2)}`);
}
grades(["10", "3.00", "2.99", "5.68", "3.01", "4", "4", "6.00", "4.50", "2.44", "5"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/03.additional/04.grades.py
students_count = int(input())

counter_top_grades = 0
counter_grades_between_4_and_5 = 0
counter_grades_between_3_and_4 = 0
counter_low_grades = 0
sum_of_grades = 0

for i in range(students_count):
    current_grade = float(input())
    sum_of_grades += current_grade
    if current_grade > 4.99:
        counter_top_grades += 1
    elif current_grade > 3.99:
        counter_grades_between_4_and_5 += 1
    elif current_grade > 2.99:
        counter_grades_between_3_and_4 += 1
    else:
        counter_low_grades += 1
        
top_grades_percentage = counter_top_grades / students_count * 100
grades_between_4_and_5_percentage = counter_grades_between_4_and_5 / students_count * 100
grades_between_3_and_4_percentage = counter_grades_between_3_and_4 / students_count * 100
low_grades_percentage = counter_low_grades / students_count * 100
average_grades = sum_of_grades / students_count

print(f'Top students: {top_grades_percentage:.2f}%')
print(f'Between 4.00 and 4.99: {grades_between_4_and_5_percentage:.2f}%')
print(f'Between 3.00 and 3.99: {grades_between_3_and_4_percentage:.2f}%')
print(f'Fail: {low_grades_percentage:.2f}%')
print(f'Average: {average_grades:.2f}')
